jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManagerTest.java

```diff
@@ -1333,6 +1333,43 @@ public void testErrorThrowOfChildProcessOnParent() {
         manager.close();
     }
     @Test
+    public void testSignalProcessAndCheckCleanup() {
+        RuntimeEnvironment environment = RuntimeEnvironmentBuilder.Factory.get()
+                .newDefaultBuilder()
+                .userGroupCallback(userGroupCallback)
+                .addAsset(ResourceFactory.newClassPathResource("BPMN2-IntermediateCatchEventSignalWithRef.bpmn2"), ResourceType.BPMN2)
+                .get();
+
+        manager = RuntimeManagerFactory.Factory.get().newPerProcessInstanceRuntimeManager(environment);
+        assertNotNull(manager);
+        RuntimeEngine runtime1 = manager.getRuntimeEngine(ProcessInstanceIdContext.get());
+        KieSession ksession1 = runtime1.getKieSession();
+
+        ksession1.startProcess("IntermediateCatchEventWithRef");
+
+
+        List<? extends ProcessInstanceLog> logs = runtime1.getAuditService().findActiveProcessInstances("IntermediateCatchEventWithRef");
+
+        assertEquals(1, logs.size());
+        manager.disposeRuntimeEngine(runtime1);
+
+        manager.signalEvent("Signal1", null);
+
+        runtime1 = manager.getRuntimeEngine(ProcessInstanceIdContext.get());
+        logs = runtime1.getAuditService().findActiveProcessInstances("IntermediateCatchEventWithRef");
+        assertEquals(0, logs.size());
+        manager.disposeRuntimeEngine(runtime1);
+        manager.close();
+
+        String pu = ((InternalRuntimeManager) manager).getDeploymentDescriptor().getPersistenceUnit();
+        EntityManagerFactory emf = EntityManagerFactoryManager.get().getOrCreate(pu);
```

Comment:
This code is not really needed, you can make use of the member attribute of the class already existing `emf` [here](https://github.com/kiegroup/jbpm/blob/4c086ba5bae070b61301edd3e8f69e61c855bc4f/jbpm-runtime-manager/src/test/java/org/jbpm/runtime/manager/impl/PerProcessInstanceRuntimeManagerTest.java#L100)

so please remove these 2 lines

Reply:
done